IntegrationTests/Services/AWSBedrockRuntimeIntegrationTests/InvokeModelWithBidirectionalStreamTest.swift

```diff
@@ -0,0 +1,233 @@
+
+        // Then, parse audio file into separate events and append them to list of events.
+        guard let audioURL = Bundle.module.url(forResource: "japan16k", withExtension: "raw") else {
+            throw ClientError.dataNotFound("Audio file not found.")
+        }
+        let audioData = try Data(contentsOf: audioURL)
+        let chunkSize = 1024
+        let totalSize = audioData.count
+        var offset = 0
+
+        while offset < totalSize {
+            let end = min(offset + chunkSize, totalSize)
+            let chunk = audioData.subdata(in: offset..<end)
+            let encodedChunk = chunk.base64EncodedString()
+            requestEvents.append(String(format: AUDIO_EVENT, encodedChunk))
+            offset += chunkSize
+        }
+    }
+
+    func testInvokeModelWithBidirectionalStream() async throws {
+        // Initialize input stream & continuation for the API input.
+        let inputStream: AsyncThrowingStream<BedrockRuntimeClientTypes.InvokeModelWithBidirectionalStreamInput, Swift.Error>
+        let continuation: AsyncThrowingStream<BedrockRuntimeClientTypes.InvokeModelWithBidirectionalStreamInput, Swift.Error>.Continuation
+        (inputStream, continuation) = AsyncThrowingStream.makeStream()
+
+        // Start up background task that feeds events to the stream.
+        Task {
+            for event in requestEvents {
+                let currentEvent = BedrockRuntimeClientTypes.InvokeModelWithBidirectionalStreamInput.chunk(
+                    BedrockRuntimeClientTypes.BidirectionalInputPayloadPart(bytes: event.data(using: .utf8))
+                )
+                // Put 0.5 second delays to simulate real time events.
+                try await Task.sleep(nanoseconds: UInt64(500_000_000))
+                continuation.yield(currentEvent)
+            }
+            continuation.yield(BedrockRuntimeClientTypes.InvokeModelWithBidirectionalStreamInput.chunk(
+                BedrockRuntimeClientTypes.BidirectionalInputPayloadPart(bytes: CONTENT_END_EVENT.data(using: .utf8))
+            ))
+            continuation.yield(BedrockRuntimeClientTypes.InvokeModelWithBidirectionalStreamInput.chunk(
+                BedrockRuntimeClientTypes.BidirectionalInputPayloadPart(bytes: PROMPT_END_EVENT.data(using: .utf8))
+            ))
+            continuation.finish()
+        }
+
+        // Create BedrockRuntime client.
+        let bedrock = try BedrockRuntimeClient(region: region)
+
+        // Call the `invokeModelWithBidirectionalStream` API.
+        let resp = try await bedrock.invokeModelWithBidirectionalStream(input: InvokeModelWithBidirectionalStreamInput(
+            body: inputStream,
+            modelId: modelID
+        ))
+
+        // Iterate on the returned output stream for output events.
+        for try await output in resp.body! {
+            switch output {
+            case .chunk(let output):
+                XCTAssertTrue(output.bytes != nil && !output.bytes!.isEmpty)
```

Comment:
is there anything more that we can verify for test success in the response stream, or is the fact that it contains non-zero audio data the best we can do?

Reply:
(i.e. in the transcribe tests we check that the transcription streamed back matches what we expected from the audio we sent)

Reply:
Given the length and variability of what exactly the model says in response, I think asserting on byte being non-null might be the best we can do for now.